tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -368,3 +373,206 @@ def __init__(self, test_context):
+# test runs a single process in the background (a.k.a BackgroundThreadService)
+# but the other tests run multiple background processes.
+class KafkaStreamsMicroservices(KafkaStreamsTest):
+    """
+    Test KafkaStreams Microservices examples which demonstrate an order management
+    workflow, such as one might find in retail and online shopping.
+    """
+
+    topics = (
+        TopicSpec(name="orders"),
+        TopicSpec(name="warehouse-inventory"),
+        TopicSpec(name="order-validations"),
+        TopicSpec(name="payments"),
+        TopicSpec(name="customers"),
+        TopicSpec(name="platinum"),
+        TopicSpec(name="gold"),
+        TopicSpec(name="silver"),
+        TopicSpec(name="bronze"),
+    )
+
+    Example = KafkaStreamExamples.KafkaStreamsMicroservices
+
+    def __init__(self, test_context):
+        super(KafkaStreamsMicroservices,
+              self).__init__(test_context=test_context,
+                             enable_pp=False,
+                             enable_sr=False)
+
+    @cluster(num_nodes=4)
+    @matrix(jar=[
+        "EmailServiceTest", "FraudServiceTest", "InventoryServiceTest",
+        "OrderDetailsServiceTest", "OrdersServiceTest",
+        "ValidationsAggregatorServiceTest"
+    ])
+    def test_kafka_streams_microservices(self, jar):
+        example = self.create_example(jar)
+        example.start()
+        example.wait()
+        example.stop()
+
+
+# Not using any of the KafkaStreams base classes because this test
+# runs a client-side script to "reset" the app in the middle of execution
+# that other tests do not use.
+class KafkaStreamsAppReset(KafkaStreamsTest):
+    """
+    Test KafkaStreams ApplicationReset example that uses confluent's cleanUp API
+    to reset an application between runs.
+    """
+    topics = (
+        TopicSpec(name="my-input-topic"),
+        TopicSpec(name="my-output-topic"),
+        TopicSpec(name="rekeyed-topic"),
+    )
+
+    Example = KafkaStreamExamples.KafkaStreamsAppReset
+    PRODUCER = AnomalyProducer
+
+    def __init__(self, test_context):
+        super(KafkaStreamsAppReset, self).__init__(test_context=test_context,
+                                                   enable_pp=False,
+                                                   enable_sr=False)
+
+    def is_valid_msg(self, msg):
+        key = msg["key"]
+        return "record-1" in key or "record-2" in key or "record-3" in key
+
+    def _consume(self, consumer):
+        def try_cons():
+            i = 0
+            msgs = consumer.messages
+            while i < len(msgs) and not self.is_valid_msg(msgs[i]):
+                i += 1
+
+            return i < len(msgs)
+
+        wait_until(
+            try_cons,
+            timeout_sec=self._timeout,
+            backoff_sec=5,
+            err_msg=f"kafka-streams KafkaStreamsAppReset consumer failed")
+
+    @cluster(num_nodes=6)
+    def test_kafka_streams_app_reset(self):
+        example = self.create_example(False)
```

Comment:
We have a duplication here: we use names to hint whether a running process is a driver or an example but at the same time we also need to pass boolean flag which dictates if the method in question should create an example or a driver. As a result it creates a situation where somebody may write `self.create_example(True)` which is an oxymoron. If we [split KafkaStreamsInteractiveQueries](https://github.com/vectorizedio/redpanda/pull/2962/commits/1ef1a23b44f0fafa2a3533ecc3d7ac3ae1e4f1eb#r755507653) it also fixes this problem.